src/probabilities_estimators/dispersion/dispersion.jl

```diff
@@ -0,0 +1,172 @@
+
+A probability estimator based on dispersion patterns, originally used by
+Rostaghi & Azami, 2016[^Rostaghi2016] to compute the "dispersion entropy", which
+characterizes the complexity and irregularity of a time series.
+
+Relative frequencies of dispersion patterns are computed using the symbolization scheme
+`s` with embedding dimension `m` and embedding delay `τ`. Recommended parameter
+values[^Li2018] are `m ∈ [2, 3]`, `τ = 1` for the embedding, and `c ∈ [3, 4, …, 8]`
+categories for the Gaussian symbol mapping. See below for an in-depth description of
+dispersion pattern construction and usage.
+
+# Algorithm
+
+Assume we have a univariate time series ``X = \\{x_i\\}_{i=1}^N``. First, this time series
+is symbolized using `s`, which default to [`GaussianSymbolization`](@ref), which uses the
+normal cumulative distribution function (CDF) for symbolization. Other choices of CDFs are
+also possible, but Entropies.jl currently only implements [`GaussianSymbolization`](@ref),
+which was used in Rostaghi & Azami (2016). This step results in an integer-valued symbol
+time series ``S = \\{ s_i \\}_{i=1}^N``, where ``s_i \\in [1, 2, \\ldots, c]``.
+
+Next, the symbol time series ``S`` is embedded into an
+``m``-dimensional time series, using an embedding lag of ``\\tau = 1``, which yields a
+total of ``N - (m - 1)\\tau`` points, or "dispersion patterns". Because each ``z_i`` can
+take on ``c`` different values, and each embedding point has ``m`` values, there
+are ``c^m`` possible dispersion patterns. This number is used for normalization when
+computing dispersion entropy.
+
+## Computing dispersion probabilities and entropy
+
+A probability distribution ``P = \\{p_i \\}_{i=1}^{c^m}``, where
+``\\sum_i^{c^m} p_i = 1``, can then be estimated by counting and sum-normalising
+the distribution of dispersion patterns among the embedding vectors.
+
+```jldoctest dispersion_example; setup = :(using Entropies)
+julia> x = repeat([0.5, 0.7, 0.1, -1.0, 1.11, 2.22, 4.4, 0.2, 0.2, 0.1], 10);
+
+julia> c, m = 3, 5;
+
+julia> est = Dispersion(s = GaussianSymbolization(c = c), m = m);
+
+julia> probs = probabilities(x, est)
+9-element Probabilities{Float64}:
+ 0.09375000000000001
+ 0.10416666666666669
+ 0.18750000000000003
+ 0.09375000000000001
+ 0.10416666666666669
+ 0.10416666666666669
+ 0.10416666666666669
+ 0.10416666666666669
+ 0.10416666666666669
+```
+
+Note that dispersion patterns that are not present are not counted. Therefore, you'll
+always get non-zero probabilities using the `Dispersion` probability estimator.
+
+To compute (normalized) dispersion entropy of order `q` to a given `base` on the
+univariate input time series `x`, do:
+
+```jldoctest dispersion_example
+julia> entropy_renyi(x, Dispersion(normalize = true), base = 2, q = 1)
+0.6716889280681666
+```
+
+If `normalize == true`, then when used in combination with [`entropy_renyi`](@ref),
+the dispersion entropy is normalized to `[0, 1]`. Normalization is only
+defined when `q == 1`.
```

Comment:
Hmmmm, here this is something we can discuss extensively. The normalization process. In general, most entropies that have a _known total number of symbols_ can be normalized versus the entropy of uniform probability for each symbol. E.g., for ordinal patterns this is also known a-priory.

Isn't it better for us to devise a general API for normalizing entropies in this regard, instead of doing it individual for each single estimator?

Here is what I propose: define a function `alphabet_length(::ProbabilitiesEstimator)` or `word_length` or whatever information theory people call it. It returns the total number of possible words. Not just the ones that actually exist in the data. If it is not known a-priori, it returns `nothing` instead. Then, we define a function `entropy_normalized` that returns the renyi entropy but normalized (i.e., divided by `log(alphabet_length)`), i.e., normalized versus uniform distribution.

Reply:
> Isn't it better for us to devise a general API for normalizing entropies in this regard, instead of doing it individual for each single estimator?
> Here is what I propose: define a function alphabet_length(::ProbabilitiesEstimator) or word_length or whatever information theory people call it. It returns the total number of possible words. Not just the ones that actually exist in the data. If it is not known a-priori, it returns nothing instead. Then, we define a function entropy_normalized that returns the renyi entropy but normalized (i.e., divided by log(alphabet_length)), i.e., normalized versus uniform distribution.

I like that idea a lot. I'll implement that in a PR with all my responses to your comments here.